packages/data/src/reviews/controls.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { controls as dataControls } from '@wordpress/data-controls';
+
+import apiFetch from '@wordpress/api-fetch';
+
+export const fetchWithHeaders = ( options ) => {
+	return {
+		type: 'FETCH_WITH_HEADERS',
+		options,
+	};
+};
+
+const controls = {
+	...dataControls,
+	FETCH_WITH_HEADERS( { options } ) {
+		let headers;
+		return apiFetch( { ...options, parse: false } )
+			.then( ( response ) => {
+				headers = response.headers;
+				return response.json();
+			} )
+			.then( ( data ) => ( { headers, data } ) );
```

Comment:
I guess its probably personal preference to not declare a variable outside the scope and manage it this way, but you could do this differently via Promise.all ([which is happy to take promise and non promise values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)).

e.g.

```
return apiFetch( { ...options, parse: false } )
			.then( ({headers, json} ) => {
                 return Promise.all([headers,json()])
			} )
			.then( ( [headers,data] ) => ( { headers, data } ) );
```

Reply:
Nice! I was thinking there must be better way to do this. I applied your suggestion in https://github.com/woocommerce/woocommerce-admin/pull/4941/commits/91a8467fc9fa4a6ce0fb5a01505aacc0de933548.

Interesting to note that destructuring of `response` led to `TypeError: Failed to execute 'json' on 'Response': Illegal invocation`. Instead, `response.json()` worked.